src/graph.py

```diff
@@ -85,16 +88,17 @@ def has_node(self, n):
     def neighbors(self, n):
         """Return a list of nodes that have edge connect to n."""
         try:
-            return self._dict[n]
+            return [x[0] for x in self._dict[n]]
         except KeyError as e:
             e.agrs = ('Node not in the graph',)
             raise
 
     def adjacent(self, n1, n2):
         """Check if 2 nodes has connection."""
         try:
-            return n2 in self._dict[n1] or\
-             n1 in self._dict[n2]
+            n1_neighbors = [x[0] for x in self._dict[n1]]
+            n2_neighbors = [x[0] for x in self._dict[n2]]
```

Comment:
This is not quite right: `adjacent` should return `True` if `n2` is in the neighbors of `n1`; it should NOT return `True` because `n1` is in the neighbors of `n2`.